Fix: forgot to add tda without beta to list of workflows :[

k-dominik opened the pull request:
<!-- Thank you very much for opening a PR!

     Please summarize your your pull request in 1-2 sentences. -->

## Checklist

<!-- Checking off an item means that an action has been successfully completed.
     Some items might not be applicable - you can remove those if you decide
     that this is the case.
-->

- [x] Format code and imports.
- [x] Add docstrings and comments.
- [x] Add tests.
- [ ] Update [user documentation](https://github.com/ilastik/ilastik.github.io).
- [ ] Reference relevant issues and other pull requests.
- [x] Rebase commits into a logical sequence.

btbest commented:
We also forgot that this breaks existing projects with this workflow... 
![image](https://github.com/user-attachments/assets/6e835cb5-8c0c-40f4-81e0-5b6ccac6e665)

k-dominik commented:
Bummer, thanks for checking this @btbest

k-dominik commented:
Hey @btbest, could you please take another look? I couldn't bring myself to keep the `workflowName` as `Trainable Domain Adaptation (Local) (beta)`. Instead I added a shallow Legacy class that enables backwards compatibility. Tiny plus is that it is also visible in the recent project list even after opening them...

<img width="376" alt="image" src="https://github.com/user-attachments/assets/1e579683-8e52-49ce-a2d1-7681f374eff4" />

btbest commented:
I love it, that's actually a really smooth solution imo 👍

Maybe even a naggy `workflowDisplayName = 'Trainable Domain Adaptation (Local) (beta - please upgrade this project file with "Import Project")'` :grin: (not sure if I'm serious about this suggestion, Import Project is a pretty obscure feature afterall...)

k-dominik commented:
> Jokes aside, considering that it does still pop up in the Other Workflows list, maybe the displayName should be "(beta / deprecated)". Or alternatively exclude it from the Other Workflows.

I think it should just work - workflow not listed if the `auto_register` thing is false. So I decluttered the workflow-listing logic a bit. In short our workflow listing is now simplified:

All workflows that are imported and have `show_in_startup_menu == True` will show up in the menu. Whether it's in a specific group, or other depends on `ilastik.shell.ilastikShell.IlastikShell_populateWorkflows`.

Before there were multiple levels with `ilastik.workflows.AVAILABLE_WORKFLOWS` that could implicitly override that property. I removed that global variable.

btbest commented:
Beautiful refactoring, love that this also simplifies some tests.

There's always more of course. The drop-down on Import Project still lists the beta variant, because it goes through `getAvailableWorkflows`, which collects all `Workflow` subclasses 🤦 But I think I would cut off here and accept that as it is, unless you think simply filtering that list by the new `show_in_startup_menu` property is sufficient. A user who wants to upgrade a "(beta)" to the non-beta variant already knows what they want to choose here after all.

![image](https://github.com/user-attachments/assets/00e9a1ef-5cd7-4e13-9519-f21f29a2d086)

emilmelnikov commented on ilastik/workflow.py:
```diff
@@ -38,8 +36,8 @@ class Workflow(Operator):
-    #: Should workflow be automatically added to start widget
-    auto_register = True
+    #: Should workflow added to start widget
+    show_in_startup_menu = True
```
Hmm, if workflows in the startup menu are explicitly listed and grouped in `_populateWorkflows()` anyway, is this flag really necessary?

k-dominik replied:
yes, this is currently the case only for the "old" name of the trainable domain adaptation workflow, but in future, we can remove the PC+OC workflow that way and still enable headless running... Which I think is a nice property.
